Release process updates observed during v1.3.0-alpha.0

SgtCoDFish opened the pull request:
Signed-off-by: Ashley Davis <ashley.davis@jetstack.io>

maelvls commented on content/en/docs/contributing/release-process.md:
````diff
@@ -15,11 +15,11 @@ perform a cert-manager release:
 2. Install our [`cmrel`](https://github.com/cert-manager/release) CLI:
    ```sh
-   (cd && GO111MODULE=on go get github.com/cert-manager/release/cmd/cmrel@latest)
+   (go get github.com/cert-manager/release/cmd/cmrel@latest)
````
```suggestion
   go install github.com/cert-manager/release/cmd/cmrel@master
```

maelvls commented on content/en/docs/contributing/release-process.md:
````diff
@@ -15,11 +15,11 @@ perform a cert-manager release:
 1. Install the [`release-notes`](https://github.com/kubernetes/release/blob/master/cmd/release-notes/README.md) CLI:
    ```sh
-   (cd && GO111MODULE=on go get k8s.io/release/cmd/release-notes@v0.7.0)
+   (go get k8s.io/release/cmd/release-notes@v0.7.0)
````
```suggestion
   go install k8s.io/release/cmd/release-notes@v0.7.0
```

maelvls commented on content/en/docs/contributing/release-process.md:
```diff
@@ -122,7 +124,7 @@ The process for cutting a minor release is as follows:
         export END_REV=release-1.0
         export START_REV=v0.16.1
-        (cd && GO111MODULE=on go get k8s.io/release/cmd/release-notes@v0.7.0)
+        (go get k8s.io/release/cmd/release-notes@v0.7.0)
```
```suggestion
        go install k8s.io/release/cmd/release-notes@v0.7.0
```

maelvls commented on content/en/docs/contributing/release-process.md:
```diff
@@ -203,16 +206,16 @@ The process for cutting a minor release is as follows:
 
         Note: At this stage there will be a draft release on GitHub and a
-        live release on [ArtifactHub][]. So you must now complete the
+        live release on [ChartMuseum][]. So you must now complete the
```
```suggestion
        live release on our [chartmuseum instance][chartmuseum]. So you must now complete the
```

maelvls commented on content/en/docs/contributing/release-process.md:
```diff
@@ -203,16 +206,16 @@ The process for cutting a minor release is as follows:
+        live release on [ChartMuseum][]. So you must now complete the
         release process quickly otherwise users of the latest release on
-        [ArtifactHub][] will encounter errors, because the manual CRD
+        [ChartMuseum][] will encounter errors, because the manual CRD
```
```suggestion
        the [chartmuseum instance][chartmuseum] will encounter errors, because the manual CRD
```

maelvls commented on content/en/docs/contributing/release-process.md:
```diff
@@ -291,4 +294,4 @@ The process for cutting a patch release is as follows:
 2. Then, continue with the instructions in [process for releasing a minor version](#process-for-releasing-a-minor-version).
 ## Links
```
I propose to remove this section since it has nothing in it:

![empoty](https://user-images.githubusercontent.com/2195781/112636520-790d5c00-8e3d-11eb-90fb-c1ce3e431804.png)

```suggestion
```

maelvls commented on content/en/docs/contributing/release-process.md:
```diff
@@ -291,4 +294,4 @@ The process for cutting a patch release is as follows:
 ## Links
 
- [ArtifactHub]: https://charts.jetstack.io
+ [ChartMuseum]: https://charts.jetstack.io
```
```suggestion
[chartmuseum]: https://charts.jetstack.io
```

SgtCoDFish replied:
I stuck with "ChartMuseum" since that seems to be how it's rendered on their site: https://chartmuseum.com/ - but I tried to be consistent with usage.

maelvls commented on content/en/docs/contributing/release-process.md:
````diff
@@ -105,8 +107,8 @@ The process for cutting a minor release is as follows:
+     the GitHub project to be able to push to the release branch.
 
      ```bash
      git push --set-upstream origin
````
```suggestion
     git push --set-upstream origin release-1.0
```

SgtCoDFish commented:
I believe I've now addressed all the comments that were previously raised, and fixed the failing build (which needed "ChartMuseum" adding to the dictionary for the spellcheck)

maelvls commented:
Thanks!!
/lgtm
